libs/template/experimental/virtual-scrolling/src/lib/observe-element-size.ts

```diff
@@ -0,0 +1,32 @@
+import { Observable } from 'rxjs';
+
+export function observeElementSize(
+  element: Element,
+  config?: {
+    options?: ResizeObserverOptions;
+  }
+): Observable<DOMRectReadOnly>;
+export function observeElementSize<T>(
+  element: Element,
+  config?: {
+    options?: ResizeObserverOptions;
+    extract: (entries: ResizeObserverEntry[]) => T;
+  }
+): Observable<T>;
+export function observeElementSize<T>(
+  element: Element,
+  config?: {
+    options?: ResizeObserverOptions;
+    extract?: (entries: ResizeObserverEntry[]) => T;
+  }
+): Observable<T | DOMRectReadOnly> {
+  const extractProp: (entries: ResizeObserverEntry[]) => T | DOMRectReadOnly =
+    config?.extract ? config?.extract : (entries) => entries[0].contentRect;
```

Comment:
Could also use null coalescing here:

```suggestion
    config?.extract ?? (entries) => entries[0].contentRect;
```